Fixed federator error on creating of a new transaction

coolwather opened the pull request.

pmprete commented on federator/src/lib/FederatorERC.ts:
```diff
@@ -415,38 +415,40 @@ export default class FederatorERC extends Federator {
-            },
-          }),
+      if (receipt !== null) {
+        if (!receipt.status) {
```
i would join both conditions into one if

pmprete commented on federator/src/lib/TransactionSender.ts:
```diff
@@ -44,17 +44,21 @@ export class TransactionSender {
+      // To fix it, we decided to use a hardcoded gas estimation
+      return +estimatedGas < ESTIMATED_GAS ? ESTIMATED_GAS : +estimatedGas;
+    } catch (err) {
+      return null;
```
This looks dangerous,  as there is an actual error here and it's being swallowed

pmprete commented on federator/src/lib/TransactionSender.ts:
```diff
@@ -136,14 +140,21 @@ export class TransactionSender {
+    const gasLimit = await this.getGasLimit(rawTx);
+
+    if (gasLimit === null) {
+      return null;
```
Instead of asking for null gas we should fix the error that is being caused on calculating the gas

pmprete commented on federator/src/lib/TransactionSender.ts:
```diff
@@ -206,6 +217,9 @@ export class TransactionSender {
     try {
       const from = await this.getAddress(privateKey);
       rawTx = await this.createRawTransaction(from, to, data, value);
+      if (rawTx === null) {
```
Why rawTx is null?

pmprete reviewed (changes requested):
There is an underlying error, instead of returning null we should fix the actual error

coolwather commented:
I removed all my previously changes in the PR, and added a validation to process the transaction vote and so on only if the transaction it's for that federator side chain.

pmprete reviewed (approved):
Good work!
